src/recon.rs

```diff
@@ -2676,12 +2676,11 @@ pub(crate) unsafe fn rav1d_recon_b_intra<BD: BitDepth>(
                             (*t_dim).w as c_int,
                             (*t_dim).h as c_int,
                             seq_hdr.intra_edge_filter,
-                            &mut BD::select_mut(&mut t.scratch.c2rust_unnamed_0.interintra_edge)
-                                .0
-                                .edge,
-                            128,
+                            edge_array,
+                            edge_offset,
                             BD::from_c((*f).bitdepth_max),
                         );
+                        let edge = edge_array.as_ptr().add(edge_offset);
```

Comment:
```suggestion
                        let edge = edge_array[edge_offset..].as_ptr();
```

Although if this doesn't work due to multiple `&mut`s, your way is okay.

Reply:
Both ways work. My thinking was that `edge` is a pointer to the middle of a slice and will be later dereferenced with negative offsets. It doesn't seem "right" to first construct a slice `[edge_offset..]` that doesn't contain elements that will be referenced by the `edge` pointer.

Reply:
Oh, that's a very good point.  I didn't realize `edge` will be accessed with negative offsets.  Your way is better in this case then.